lib/install/validate/storage.go

```diff
@@ -174,3 +184,61 @@ func (v *Validator) suggestDatastore(path string, label string, flag string) {
 		}
 	}
 }
+
+// checks if a given datastore ds is writeable by compute resource specified on cmd line --compute-resource
+func (v *Validator) targetDatastoreIsWriteable(ctx context.Context, ds *object.Datastore) bool {
+	defer trace.End(trace.Begin(ds.String()))
+
+	// get compute hosts if we haven't run this method on this object before
+	if len(v.computeHosts) == 0 {
+		var err error
+		v.computeHosts, err = v.Session.Cluster.Hosts(ctx)
+		if err != nil {
+			v.NoteIssue(err)
+			return false
+		}
+		// v.computeHosts shouldn't be zero anymore; if it still is, there's a problem
+		if len(v.computeHosts) == 0 {
+			v.NoteIssue(errors.New("Couldn't get list of hosts connected to target cluster"))
+			return false
+		}
+	}
```

Comment:
Can this whole `if` block be made a helper?  It smells like it's probably used elsewhere and a helper called in the constructor is probably the right thing to do vs doing it everywhere we need a host.